src/util/address.rs

```diff
@@ -1510,4 +1514,16 @@ mod tests {
         assert_eq!(Address::from_script(&bad_p2wsh, Network::Bitcoin), expected);
         assert_eq!(Address::from_script(&invalid_segwitv0_script, Network::Bitcoin), Err(Error::InvalidSegwitV0ProgramLength(17)));
     }
+
+    #[test]
+    fn test_address_type_parsing() {
+        // Test successful parsing of a valid `AddressType`
+        assert_eq!(AddressType::from_str("p2tr"), Ok(AddressType::P2tr));
+        // Test that parsing an invalid `AddressType` fails.
+        assert_eq!(
+            AddressType::from_str("invalid"),
+            Err(Error::InvalidAddressType("invalid".to_string()))
+        );
+    }
+
```

Comment:
Admittedly our test practices are a bit loosey-goosey but this one could easily be made super clean by doing
```rust

    #[test]
    fn valid_address_parses_correctly() {
        let addr = AddressType::from_str("p2tr").expect("false negative while parsing address");
        assert_eq!(addr, AddressType::P2tr);
    }

    #[test]
    fn invalid_address_parse_error() {
        let got = AddressType::from_str("invalid");
        let want = Err(Error::InvalidAddressType("invalid".to_string()));
        assert_eq!(got, want);
    }
```

Reply:
Bad github-foo, I did not add enough context when doing this suggestion, my bad.

Reply:
Got you! Yeah, this is done too.

Reply:
I was actually thinking about iterating over the valid address types in the test, but didn't do it in the end because I thought it to be slightly over-kill. Wdyt? I've left it as you suggested, but happy to add that if it seems worth it! :)

Reply:
Tempting but not sure if worth it either.